src/FSharp.Data.GraphQL.Server.AspNetCore/GraphQLOptions.fs

```diff
@@ -0,0 +1,37 @@
+namespace FSharp.Data.GraphQL.Server.AspNetCore
+
+open FSharp.Data.GraphQL
+open System
+open System.Text.Json
+open System.Threading.Tasks
+open Microsoft.AspNetCore.Http
+
+type PingHandler = IServiceProvider -> JsonDocument option -> Task<JsonDocument option>
+
+type GraphQLTransportWSOptions = {
+    EndpointUrl : string
+    ConnectionInitTimeoutInMs : int
+    CustomPingHandler : PingHandler option
+}
+
+type IGraphQLOptions =
+    abstract member SerializerOptions : JsonSerializerOptions
+    abstract member WebsocketOptions : GraphQLTransportWSOptions
+    abstract member GetSerializerOptionsIdented : unit -> JsonSerializerOptions
+
+type GraphQLOptions<'Root> = {
+    SchemaExecutor : Executor<'Root>
+    RootFactory : HttpContext -> 'Root
+    SerializerOptions : JsonSerializerOptions
+    WebsocketOptions : GraphQLTransportWSOptions
+} with
+
+    member options.GetSerializerOptionsIdented () =
+        let options = JsonSerializerOptions (options.SerializerOptions)
+        options.WriteIndented <- true
+        options
```

Comment:
Do you have a specific use case in mind?
Just to remember, you can do customizations to `SerializerOptions` when adding `GraphQLOptions<'Root>` to the program's IoC container. Example:
```fsharp
[<EntryPoint>]
let main args =

    let builder = WebApplication.CreateBuilder (args)
    builder.Services
        .AddGiraffe()
        .AddGraphQLOptions<Root> (
            Schema.executor,
            rootFactory,
            "/ws",
            fun jsonOptions ->
                jsonOptions.SerializerOptions.WriteIndented <- true
                jsonOptions
        )
    |> ignore

    let app = builder.Build ()
    ...
```

Reply:
But I only want that for logging purposes and only if the `Trace` level set

Reply:
Ok, we discussed that further and came to an agreement. Resolving this conversation...